app/api/groups/get_team_descendants.go

```diff
@@ -102,10 +103,15 @@ func (srv *Service) getTeamDescendants(w http.ResponseWriter, r *http.Request) s
 	service.MustNotBeError(srv.Store.Groups().
 		Select("parent_links.child_group_id AS linked_group_id, groups.id, groups.name").
 		Joins(`
-			JOIN groups_groups AS parent_links ON parent_links.parent_group_id = groups.id AND
-				parent_links.type = 'direct' AND parent_links.child_group_id IN (?)`, groupIDs).
+			JOIN groups_groups AS parent_links
+			ON parent_links.parent_group_id = groups.id AND
+				parent_links.type = 'direct' AND
+				NOW() < parent_links.expires_at AND
+				parent_links.child_group_id IN (?)`, groupIDs).
 		Joins(`
-			JOIN groups_ancestors AS parent_ancestors ON parent_ancestors.child_group_id = groups.id AND
+			JOIN groups_ancestors AS parent_ancestors
+			ON parent_ancestors.child_group_id = groups.id AND
+				NOW() < parent_ancestors.expires_at AND
```

Comment:
When I see all these `NOW() < parent_ancestors.expires_at ` added, I wonder if it wouldn't have been wiser to create views `groups_ancestors_notexpired` and `groups_groups_notexpired` and to use them across the requests. 
These conditions and polluting sometimes-already-complex requests and could easily be forgotten and removed by mistake. 
There are no test updates for them :-/ and probably it would have been too much to add tons of tests but then actually a mistake would be so easy here.  I think when there are too many repetitive changes for the same things to justify tests (and because testings ancestors membership expiration would be out of scope of the function), maybe it is good place to hide the "logic" into something that could be considered as tested separately. I am afraid of mistakes on this expiration (not necessarily now but more regressions). 
What are your thoughts about that?

Reply:
Fixed. Let's see if it is better with the views...